enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+Given the high source of engineering effort needed to accomplish the eventual delegation of optional, CVO-based operators to OLM over time, this first phase is centered around adding basic support for installing POs and impacting the cluster lifecycle in some fashion.
+
+**Note**: This phase may require manual upgrades of OLM-based operators installed through this PO mechanism. The process for manually upgrading PO packages will be detailed below.
+
+In addition, future phase requirements around package dependency resolution may require a new design entirely. While this is a notable short term trade-off, it allows the OLM team to continue iterating on their "deppy" component in the background. Deppy is an OLM 1.x component that's responsible for dependency resolution. A future design may elect to use deppy and it's APIs, and delegate package selection to that component. The usage of feature flags can be used to toggle the underlying implementation layer once that component is ready.
+
+This phase will likely restrict which OLM-based operators admins can install. Those requirements will be detailed below. The immediate pool of candidates may include the cert-manager and service-binding-operator operators.a
+
+#### Phase 1: Expand Capabilities
+
+Summary: Introduce depedency resolution, and expand the set of supported capabilities for admins and platform operator authors. Help establish, or improve, any first-class APIs at the installer and cluster runtime levels.
+
+The previous phase aimed to get the foundational layer in place to start onboarding more OLM-based operators to participate in the cluster's lifecycle. The notable limitations with phase 0 is requiring manual admin intervention around editing the installer's Kubernetes manifests and upgrading PO packages after installation. This phase will aim to address those limitations by introducing first-class concepts.
+
+This phase will also attempt to introduce package management and dependency resolution primitives. Candidate PO packages may place dependencies on other packages and GVKs, in addition to more granular constraints that are needed in order to accommodate EUS-EUS drift and platform-specific cluster invariants.
+
+#### Phase 2: QoL Improvements
+
+Summary: Prepare for general availability by tackling any performance improvements, shoring up the overall design/implementation/UX, and any QoL improvements that have been identified from previous phase feedback.
+
+This phase has largely been unexplored. Candidates for this phase may include:
+
+- Integration with the OCP console, hypershift, KCP, etc.
+- Improved instrumentation.
+- Improved performance of the overall stack.
+- Improved UX around the user-facing APIs.
+
+### PO Requirements
+
+- Must package Kubernetes manifests using OLM's registry+v1 bundle format.
```

Comment:
Yep, that's a good call out. There are a couple of reasons for this bundle format requirement, but it's mainly because our newest "plain bundle" format isn't ready for production usage yet, and migrating the entire catalog ecosystem away from the legacy registry+v1 bundle format is untenable right now.

There's also external team dependencies here too, like the downstream pipelines that produce the catalog images, that wouldn't be able to accommodate a new bundle format in the short term, and OLM's runtime component isn't in a position where it can handle "hybrid" (i.e. contains multi-format bundles) catalogs yet.

Reply:
I ended up moving the requirements section to live in the proposal in the last couple of commits, and added some commentary around why these restrictions are needed for phase 0.